pkg/artifactory/resource/security/resource_artifactory_group.go

```diff
@@ -260,15 +260,14 @@ func (r *ArtifactoryGroupResource) Read(ctx context.Context, req resource.ReadRe
 		SetResult(&group).
 		Get(GroupsEndpoint + data.Id.ValueString())
 
-	if err != nil {
-		unableToRefreshResourceError(resp, err)
-		return
-	}
-
 	// Treat HTTP 404 Not Found status as a signal to recreate resource
 	// and return early
-	if response.StatusCode() == http.StatusBadRequest || response.StatusCode() == http.StatusNotFound {
-		resp.State.RemoveResource(ctx)
+	if err != nil {
+		if response.StatusCode() == http.StatusBadRequest || response.StatusCode() == http.StatusNotFound {
```

Comment:
We should only check for 404 for groups: https://jfrog.com/help/r/jfrog-rest-apis/get-group-details?tocId=iLr3VCaSAoke_By81l_HXw

Reply:
Removed.